### ISSUE TITLE
Serial-over-LAN and raspberry?

### ISSUE THREAD

**Issue by yeahman45** [kk7ds/pynx584#58]
Not really related to the library itself but does anyone know how to setup serial over lan and raspberry connected to the NX584?
I mean what cable should I buy.. 
will this cable work?
https://www.aliexpress.com/item/32881078519.html
or this one?
https://www.aliexpress.com/item/4000564126588.html

my raspberry pi is connected via ethernet....how can I use both ethernet for networking and ethernet of the serial over lan cable?

**Comment by Pre10der** [comment]
Personally, I use a standard Cisco roll-over cable with my Keyspan USB to serial adaptor (https://www.tripplite.com/keyspan-high-speed-usb-to-serial-adapter~USA19HS). My RPi is located next to the the panel, so the ethernet endpoint is the RPi itself, not the serial adapter.

It looks like this:
NX584  <-->  DB9_SERIAL_CABLE_RJ45 <--> RJ45_DB9_Adapter  <--> USB_SERIAL_ADAPTER <-->  RPi running NX584 Server  <--> LAN <--> Home Assistant Running NX584 Client 

The cables you have listed are for carrying serial communications, not ethernet, so they are for connecting between the NX584 card, and the serial adaptor itself. If you want to use a true Ethernet to serial adaptor (for having your RPI in a different location to the panel), then you will still need those cables, but also a serial to ethernet adaptor....but I have not tested any of these so cannot recommend one sorry.

**Comment by yeahman45** [comment]
https://www.tripplite.com/keyspan-high-speed-usb-to-serial-adapter~USA19HS
it is a male serial... the serial port on the NX584 is also male serial. I guess you use a male to female adapter?
My raspberry pi is not next to the panel.. it is about 7m away.. a serial to usb cable might not be long enough... will try serial over lan adapters/extenders and see if it works.
I need to make my NX584 work with my nx-4 panel first; I failed making it work last time I tried; messed up my alarm system.

**Comment by Pre10der** [comment]
Yes, you are right....forgot I had an RJ45 to DB9 adapter in the mix as well....its been like that for over 2yrs now and I must have just used what I had lying around. Updated original comment to reflect that. Learning to play with the panels is fun...I defaulted mine entirely and worked to set it up bit by bit until I got it right. Document as you go....it helps for when you come back to it some time in the future.

**Comment by yeahman45** [comment]
thx for your reply.
So that "extender" should work?
https://www.aliexpress.com/item/4000564126588.html
seen this post https://github.com/kk7ds/pynx584/issues/48 mentioning some cables do not work?

"I defaulted mine entirely and worked to set it up bit by bit until I got it right. Document as you go....it helps for when you come back to it some time in the future."
Not sure I have the skills to configure everything from scratch.
Did you install the NX584 on your panel while it is powered on? last time I powered down my alarm when installing the NX584 but still could not get it enrolled :(

**Comment by Pre10der** [comment]
It should work if you are just looking to make a long serial cable using CAT5e/6 cable providing the pinout is correct. The pinout is the important bit....the other thread you posted in spoke to the correct pinout (it is a physical thing, not a software thing). This document speaks to it also: http://www.increa.com/reverse/nx-8e/Caddx_NX-584_Communication_Protocol.pdf

You will need to know the 'installer' (master) code of your panel to enrol the NX584 - that is, not just the standard user code you use to arm/disarm. If you did not install the panel, chances are you won't have the installer code as most installers do not give that away - provides an opportunity for repeat business. 

That is why I was forced to default my panel as I did not have the 'installer' code.

**Comment by yeahman45** [comment]
is the installer code the same as the master code? I think I have it.. it is the default code.

Side issue with my nx-4:
I have the service led constantly on (does not blink; solid light).
Pressing * and then 2 (as explained in https://alarm.net/wp-content/uploads/2018/08/Troubleshooting-Guide-NX-Tech-Support.pdf), I have the number 7 illuminated.
This means :
```
Failure to communicate - this light will light up when there is a failure to communicate between your system and the control station. To clear a successful signal most be received at Central Station.
```

But my alarm is not connected to any control station.
I received phone calls to my mobile phone when the alarm is triggered though.

Any idea how to remove this error? I am suspecting this error is disrupting the enrollment process for my NX584.

**Comment by Pre10der** [comment]
It would be wise to get hold of an installation manual for your alarm panel, or seek some advice from a suitable forum. I am no alarm tech, but reading the doco, it says that "This led can only be extinguished by communicating to the base. Check Phone No, Account Code and Format programming." You could try disabling Feature 9, Segment 2 (Fail to communicate).
I find the NX595E module with the DLX900 windows software a much easier way to program the panels FWIW - you can also use it to backup your current config in case you need to rollback. 

Nb. The 595E is not really useful in non-proprietary home automation setups (incl. this library) as it uses proprietary mobile apps like Ultrasync....but then again, they may be your best way to go? Really depends on what you are wanting to do.

**Comment by yeahman45** [comment]
> The pinout is the important bit....the other thread you posted in spoke to the correct pinout (it is a physical thing, not a software thing). This document speaks to it also: http://www.increa.com/reverse/nx-8e/Caddx_NX-584_Communication_Protocol.pdf

but how can I find out if the pinout is correct before ordering? It seems sellers don't put that kind of information

**Comment by yeahman45** [comment]
lol still confused what kind of DB9 to usb adapter to buy concerning the pinouts..will this one work:
https://www.aliexpress.com/item/4000483271733.html
CH340 chipset:
https://static.efetividade.net/img/ch340g-datasheet-34852.pdf
I have no idea how to read the schematic lol

**Comment by kk7ds** [comment]
Can I close this now? Sounds like this is not a pynx584 issue, but a wiring one. If someone wants to PR some documentation about cables, adapters, and pinouts that's cool, but I think we can probably close this out to get it off the books, right?
